agent-control/src/values.rs

```diff
@@ -1,7 +1,147 @@
+use std::sync::Arc;
+
+use opamp_client::operation::capabilities::Capabilities;
+use tracing::debug;
+
+use crate::{
+    agent_control::{
+        agent_id::AgentID,
+        defaults::{STORE_KEY_LOCAL_DATA_CONFIG, STORE_KEY_OPAMP_DATA_CONFIG},
+    },
+    opamp::{data_store::OpAMPDataStore, remote_config::hash::ConfigState},
+    values::{
+        config::{Config, RemoteConfig},
+        config_repository::{ConfigRepository, ConfigRepositoryError},
+        yaml_config::{YAMLConfig, has_remote_management},
+    },
+};
+
 pub mod config;
 pub mod config_repository;
+pub mod yaml_config;
 
-pub mod file;
+pub struct ConfigRepo<D: OpAMPDataStore> {
+    opamp_data_store: Arc<D>,
+    remote_enabled: bool,
+}
 
-pub mod k8s;
-pub mod yaml_config;
+impl<D: OpAMPDataStore> ConfigRepo<D> {
+    pub fn new(opamp_data_store: Arc<D>) -> Self {
+        Self {
+            opamp_data_store,
+            remote_enabled: false,
+        }
+    }
+
+    pub fn with_remote(self) -> Self {
+        Self {
+            remote_enabled: true,
+            ..self
+        }
+    }
+}
+
+impl<D> ConfigRepository for ConfigRepo<D>
+where
+    D: OpAMPDataStore + Send + Sync + 'static,
+{
+    #[tracing::instrument(skip_all, err)]
+    fn load_local(&self, agent_id: &AgentID) -> Result<Option<Config>, ConfigRepositoryError> {
+        self.opamp_data_store
+            .get_local_data::<YAMLConfig>(agent_id, STORE_KEY_LOCAL_DATA_CONFIG)
```

Comment:
Why isn't `STORE_KEY_LOCAL_DATA_CONFIG` hardcoded inside `get_local_file_path` like the `FOLDER_NAME_LOCAL_DATA`?